compiler/optimizer/InductionVariable.cpp

```diff
@@ -3643,6 +3651,79 @@ bool TR_LoopStrider::reassociateAndHoistComputations(TR::Block *loopInvariantBlo
          }
       }
 
+#ifdef J9_PROJECT_SPECIFIC
```

Comment:
I don't think this is part of just a simple disabling of the logic for data address pointer (?) If so, are you able to add some code comment descriptions for this code and others like it in this commit to explain what is being done ? Note that the commit message may also need to be updated in such case.

Reply:
The code changed without updating the commit.
The final code teaches inductionVariable about OffHeap array access tree structure and skipping examining dataAddress nodes as traditional internalPointers (trying to find the induction variable in its children).
Induction variable can hoist computations for OffHeap.